hack/make-rules/verify.sh

```diff
@@ -192,6 +206,7 @@ fi
 ret=0
 run-checks "${KUBE_ROOT}/hack/verify-*.sh" bash
 run-checks "${KUBE_ROOT}/hack/verify-*.py" python
+target-checks
```

Comment:
can we run this first to fail fast if a bad target is specified?

Reply:
The reason is  the functionality of "is-explicitly-chosen" called by run-checks  can be leveraged to resolve this issue.

Currently we retrieve target list basing on the user's input paramter of run-checks.
If we want to run it before "run-checks", we need to ensure all verification test scripts are saved under "${KUBE_ROOT}/hack" directory forever.

Basing on the current fix, the end-user could get all failure information about both invalid and failure targets. For example 

     WHAT="xx godeps imports" 
Where
    xx:          the invalid target
    godeps: test would fail

The output  is

========================
FAILED TESTS
========================
godeps
xx

Reply:
It still won't fail fast, because we don't print/exit until below this. The verify job isn't currently designed to fail fast.